src/quic/data.cc

```diff
@@ -0,0 +1,271 @@
+  return convert<uv_buf_t, char>();
+}
+
+Store::operator ngtcp2_vec() const {
+  return convert<ngtcp2_vec, uint8_t>();
+}
+
+Store::operator nghttp3_vec() const {
+  return convert<nghttp3_vec, uint8_t>();
+}
+
+void Store::MemoryInfo(MemoryTracker* tracker) const {
+  tracker->TrackField("store", store_);
+}
+
+// ============================================================================
+
+namespace {
+std::string TypeName(QuicError::Type type) {
+  switch (type) {
+    case QuicError::Type::APPLICATION: return "APPLICATION";
+    case QuicError::Type::TRANSPORT: return "TRANSPORT";
+    case QuicError::Type::VERSION_NEGOTIATION: return "VERSION_NEGOTIATION";
+    case QuicError::Type::IDLE_CLOSE: return "IDLE_CLOSE";
+  }
+  UNREACHABLE();
+}
+}  // namespace
+
+QuicError::QuicError(const std::string_view reason)
+    : reason_(reason),
+      ptr_(&error_) {}
+
+QuicError::QuicError(const ngtcp2_connection_close_error* ptr)
+    : reason_(reinterpret_cast<const char*>(ptr->reason), ptr->reasonlen),
+      ptr_(ptr) {}
+
+QuicError::QuicError(const ngtcp2_connection_close_error& error)
+    : reason_(reinterpret_cast<const char*>(error.reason), error.reasonlen),
+      error_(error),
+      ptr_(&error_) {}
+
+QuicError::operator bool() const {
+  if ((code() == NO_ERROR && type() == Type::TRANSPORT) ||
+     ((code() == APP_NO_ERROR && type() == Type::APPLICATION))) {
+    return false;
+  }
+  return true;
+}
+
+const uint8_t* QuicError::reason_c_str() const {
+  return reinterpret_cast<const uint8_t*>(reason_.c_str());
+}
+
+bool QuicError::operator!=(const QuicError& other) const {
+  return !(*this == other);
+}
+
+bool QuicError::operator==(const QuicError& other) const {
+  if (this == &other) return true;
+  return type() == other.type() &&
+         code() == other.code() &&
+         frame_type() == other.frame_type();
+}
+
+QuicError::Type QuicError::type() const {
+  return static_cast<Type>(ptr_->type);
+}
+
+QuicError::error_code QuicError::code() const {
+  return ptr_->error_code;
+}
+
+uint64_t QuicError::frame_type() const {
+  return ptr_->frame_type;
+}
+
+const std::string_view QuicError::reason() const {
+  return reason_;
+}
+
+QuicError::operator const ngtcp2_connection_close_error&() const {
+  return *ptr_;
+}
+
+QuicError::operator const ngtcp2_connection_close_error*() const {
+  return ptr_;
+}
+
+MaybeLocal<Value> QuicError::ToV8Value(Environment* env) const {
+  Local<Value> argv[] = {
+    Integer::New(env->isolate(), static_cast<int>(type())),
+    BigInt::NewFromUnsigned(env->isolate(), code()),
+    Undefined(env->isolate()),
+  };
+
+  if (reason_.length() > 0 &&
+      !node::ToV8Value(env->context(), reason()).ToLocal(&argv[2])) {
+    return MaybeLocal<Value>();
+  }
+  return Array::New(env->isolate(), argv, arraysize(argv)).As<Value>();
+}
+
+std::string QuicError::ToString() const {
+  auto str = std::string("QuicError(") + TypeName(type()) + ") " +
+             std::to_string(code());
```

Comment:
```suggestion
  std::string str = "QuicError(" + TypeName(type()) + ") " +
             std::to_string(code());
```
Can't this just be simplified to this?